custom/enikshay/data_store.py

```diff
@@ -32,10 +32,11 @@ def latest_adherence_date(self, episode_id):
         if result:
             # the result is sorted on 'adherence_date'
             latest_date = result[0].get('adherence_date')
-            if not latest_date or not parse_date(latest_date):
+            parsed_date = parse(latest_date).date()
```

Comment:
`dateutil.parser.parse` tries to fill in a lot of blanks, so for example, if you call `parse('')` with an empty string it returns the current date.

so it would be good to first validate the format of the string value to make sure it matches any of the patterns you want to support.

also note `parse()` itself can raise an exception if it can't parse the given format.